### PR TITLE
[orchagent] Add support for Innovium platform

### PR DESCRIPTION
**Pull request by tonytitus**
Signed-off-by: Tony Titus ttitus@innovium.com

**What I did**

**Why I did it**

**How I verified it**

**Details if related**

### CODE REVIEW

**Comment by prsunny** [comment]
retest this please

**Comment by yxieca** [comment]
@tonytitus Can you explain why you need this change in 201811 branch? I think new feature should stay in master unless there is a strong reason for back porting to 201811 branch.

**Comment by lguohan** [comment]
can you resolve the conflict?

**Comment by tonytitus** [comment]
> can you resolve the conflict?

Done

**Comment by lguohan** [comment]
better to do rebase. merge will show lots of commits in the pr.

**Comment by tonytitus** [comment]
> better to do rebase. merge will show lots of commits in the pr.

Done

File: orchagent/portsorch.cpp [hunk 1]
```diff
@@ -1634,7 +1634,10 @@ void PortsOrch::doPortTask(Consumer &consumer)
             {
                 for (auto it = m_portListLaneMap.begin(); it != m_portListLaneMap.end();)
                 {
-                    if (m_lanesAliasSpeedMap.find(it->first) == m_lanesAliasSpeedMap.end())
+                    sai_uint32_t port_speed;
+                    auto is = m_lanesAliasSpeedMap.find(it->first);
+                    getPortSpeed(it->second, port_speed);
+                    if ((m_lanesAliasSpeedMap.find(it->first) == m_lanesAliasSpeedMap.end()) || (get<1>(is->second) != port_speed))
```
Comment (lguohan) [inline orchagent/portsorch.cpp]: why do we need to compare the port speed here?

Comment (tonytitus) [reply orchagent/portsorch.cpp]: When number of lanes are same but speed is different in port_config.ini(and config_db.json), we still need to remove port and add port. Without this change the port will not get recreated for new speed.

Comment (lguohan) [reply orchagent/portsorch.cpp]: if lanes are the same set, then we just need to set speed. why do we need to remove and add port?

Comment (tonytitus) [reply orchagent/portsorch.cpp]: When we breakout a port from one 100G (2lane * 50G PAM4) to two 50G (2lane * 25G NRZ) then we will have to recreate the port as the FEC mode could change. Same is true when we breakout a port from one 200G (4lane * 50G PAM4) to two 100G (4lane * 25G NRZ).

Comment (lguohan) [reply orchagent/portsorch.cpp]: sai should handle the fec change accordingly, or let application handle the fec settings. such recreating port should not be expose to application.

Comment (tonytitus) [reply orchagent/portsorch.cpp]: I removed this change

**Comment by lguohan** [comment]
retest this please

**Comment by yxieca** [comment]
This PR cannot be cherry-picked into 201811 branch cleanly. Separate PR is required for 201811 branch.

**Comment by songqh2022** [comment]
What I did
Support for platforms based on Clounix Networks' device
Why I did it
Support for platforms based on Clounix Networks' device
How I verified it
Have verified on the hardware device
Details if related

**Pull request merged**